cli/src/diff_util.rs

```diff
@@ -749,6 +760,107 @@ pub fn show_diff_summary(
+    for hunk in hunks {
+        for (line_type, _content) in hunk.lines {
+            match line_type {
+                DiffLineType::Context => {}
+                DiffLineType::Removed => dels += 1,
+                DiffLineType::Added => adds += 1,
+            }
+        }
+    }
+    DiffStat {
+        path: repo_path,
+        added: adds,
+        removed: dels,
+    }
+}
+
+pub fn show_diff_stat(
+    ui: &Ui,
+    formatter: &mut dyn Formatter,
+    workspace_command: &WorkspaceCommandHelper,
+    tree_diff: TreeDiffIterator,
+) -> Result<(), CommandError> {
+    let mut stats: Vec<DiffStat> = vec![];
+    let mut max_path_length = 0;
+    let mut max_diffs = 0;
+    for (repo_path, diff) in tree_diff {
+        let path = workspace_command.format_file_path(&repo_path);
+        let mut left_content: Vec<u8> = vec![];
+        let mut right_content: Vec<u8> = vec![];
+        match diff {
+            tree::Diff::Modified(left, right) => {
+                left_content = diff_content(workspace_command.repo(), &repo_path, &left)?;
+                right_content = diff_content(workspace_command.repo(), &repo_path, &right)?;
+            }
+            tree::Diff::Added(right) => {
+                right_content = diff_content(workspace_command.repo(), &repo_path, &right)?;
+            }
+            tree::Diff::Removed(left) => {
+                left_content = diff_content(workspace_command.repo(), &repo_path, &left)?;
+            }
+        }
+        max_path_length = max(max_path_length, path.len());
+        let stat = get_diff_stat(path, &left_content, &right_content);
+        max_diffs = max(max_diffs, stat.added + stat.removed);
+        stats.push(stat);
+    }
+
+    let display_width = usize::from(ui.term_width().unwrap_or(80)) - 4; // padding
+    let max_bar_length =
+        display_width - max_path_length - " | ".len() - max_diffs.to_string().len() - 1;
+    let factor = if max_diffs < max_bar_length {
+        1.0
+    } else {
+        max_bar_length as f64 / max_diffs as f64
+    };
+    let number_padding = max_diffs.to_string().len();
+
+    formatter.with_label("diff", |formatter| {
+        let mut total_added = 0;
+        let mut total_removed = 0;
+        for stat in &stats {
+            total_added += stat.added;
+            total_removed += stat.removed;
+            let bar_added = (stat.added as f64 * factor).ceil() as usize;
+            let bar_removed = (stat.removed as f64 * factor).ceil() as usize;
+            // pad to max_path_length
+            write!(
+                formatter.labeled("stat-line"),
+                " {:<max_path_length$} | {:>number_padding$}{}",
+                stat.path,
+                stat.added + stat.removed,
+                if bar_added + bar_removed > 0 { " " } else { "" },
+            )?;
+            write!(formatter.labeled("added"), "{}", "+".repeat(bar_added))?;
+            writeln!(formatter.labeled("removed"), "{}", "-".repeat(bar_removed))?;
```

Comment:
Sorry that I didn't notice this before, but the "stat-line" label should probably apply to the whole line. Alternatively, we can drop the label for now since we're not coloring it differently anyway. I would drop it.

Reply:
> we're not coloring it differently anyway

I think we are? Isn't that were the green/red colors come from?

<img width="1082" alt="image" src="https://github.com/martinvonz/jj/assets/4950/e368f69b-65fd-4ec7-ae63-3e2889c7b285">

Reply:
Do you mean the green and red pluses and minuses? That's because of the `added` and `removed` labels, not because of the `stat-line` label, I'm pretty sure.

Reply:
I thought you meant apply `stat-line` to the whole line? Won't that make the whole thing white? Sorry, I'm not terribly familiar with the formatter...

Reply:
Ok, I after re-reading I think I understand what you mean. I dropped the labeling part...

Reply:
Ah, nope. It's kind of like a very simplified CSS. The labels added here form a tree. Add the "diff" label at the outermost level. Then you add the "added" and "removed" labels around the pluses and minuses. The formatter then uses those labels to choose the colors from here: https://github.com/martinvonz/jj/blob/c3d9ba9ca91e4a4fc415db60af0f0ec9697d0aac/cli/src/config/colors.toml#L58-L59

Looks like we don't have any documentation about that :( We should add that.